mysql/mysqldump.go

```diff
@@ -228,18 +266,29 @@ func processConstraint(conv *internal.Conv, table string, constraint *ast.Constr
 		updateCols(conv, ast.ConstraintPrimaryKey, constraint.Keys, st.ColDefs, table)
 	case ast.ConstraintForeignKey:
 		st.ForeignKeys = append(st.ForeignKeys, toForeignKeys(conv, constraint))
+	case ast.ConstraintIndex:
+		st.Indexes = append(st.Indexes, schema.Index{Name: constraint.Name, Keys: toSchemaKeys(constraint.Keys)})
+	case ast.ConstraintUniq:
+		// Convert unique column constraint in MySQL to a corresponding unique index in Spanner since
+		// Spanner doesn't support unique constraints on columns.
+		// TODO: Avoid Spanner-specific schema transformations in this file -- they should only
+		// appear in toddl.go. This file should focus on generic transformation from source
+		// database schemas into schema.go.
+		st.Indexes = append(st.Indexes, schema.Index{Name: constraint.Name, Unique: true, Keys: toSchemaKeys(constraint.Keys)})
 	default:
 		updateCols(conv, ct, constraint.Keys, st.ColDefs, table)
 	}
 	conv.SrcSchema[table] = st
 }
 
-// toSchemaKeys converts a string list of MySQL primary keys to
-// schema primary keys.
+// toSchemaKeys converts a string list of MySQL keys to schema keys.
+// Note that we map all MySQL keys to ascending ordered schema keys.
+// For primary keys: this is fine because MySQL primary keys are always ascending.
+// However, for non-primary keys (aka indexes) this is incorrect: we are dropping
+// the MySQL key order specification.
+// TODO: Resolve ordering issue for non-primary keys.
 func toSchemaKeys(columns []*ast.IndexPartSpecification) (keys []schema.Key) {
 	for _, colname := range columns {
-		// MySQL primary keys have no notation of ascending/descending.
-		// We map them all into ascending primary keys.
```

Comment:
We have updated the comment but I don't see handling of ASC/DESC order in the code below; we are also missing the corresponding test cases as well as to handle ordering.

Reply:
I remember now that we are not able to parse ASC/DESC information for mysqldump parser. Can you please add a TODO here pointing to the issue that you filed - https://github.com/cloudspannerecosystem/harbourbridge/issues/96?

Reply:
Yes added todo and linked the issue.